engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -1871,6 +1908,204 @@ public static long minus(@Nullable final ZonedDateTime dateTime1, @Nullable fina
+        if (period == null || scalar == NULL_INT) {
+            return null;
+        }
+
+        return period.multipliedBy(scalar);
+    }
+
+    /**
+     * Multiply a period by a scalar.
+     *
+     * @param period the period to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the period multiplied by the scalar
+     */
+    public static Period multiply(final int scalar, final Period period) {
+        return multiply(period, scalar);
+    }
+
+    /**
+     * Multiply a period by a scalar.
+     *
+     * @param period the period to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the period multiplied by the scalar
+     */
+    public static Period multiply(final Period period, final long scalar) {
+        if (period == null || scalar == NULL_LONG) {
+            return null;
+        }
+
+        if (scalar > Integer.MAX_VALUE || scalar < Integer.MIN_VALUE) {
+            throw new DateTimeOverflowException("Scalar value is too large to be cast to an int");
+        }
+
+        return period.multipliedBy((int) scalar);
+    }
+
+    /**
+     * Multiply a period by a scalar.
+     *
+     * @param period the period to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the period multiplied by the scalar
+     */
+    public static Period multiply(final long scalar, final Period period) {
+        return multiply(period, scalar);
+    }
+
+    /**
+     * Divide a duration by a scalar.
+     *
+     * @param duration the duration to divide
+     * @param scalar the scalar to divide by
+     * @return {@code null} if either input is {@code null}; otherwise the duration divide by the scalar
+     */
+    public static Duration divide(final Duration duration, final long scalar) {
+        if (duration == null || scalar == NULL_LONG) {
+            return null;
+        }
+
+        if (scalar == 0) {
+            throw new DateTimeOverflowException("Scalar value is zero");
+        }
+
+        return duration.dividedBy(scalar);
+    }
+
+    /**
+     * Divide a duration by a scalar.
+     *
+     * @param duration the duration to divide
+     * @param scalar the scalar to divide by
+     * @return {@code null} if either input is {@code null}; otherwise the duration divide by the scalar
+     */
+    public static Duration divide(final Duration duration, final double scalar) {
+        if (duration == null || scalar == NULL_DOUBLE) {
+            return null;
+        }
+
+        if (Double.isNaN(scalar)) {
+            throw new DateTimeOverflowException("Scalar value is NaN");
+        }
+
+        if (scalar == 0) {
+            throw new DateTimeOverflowException("Scalar value is zero");
+        }
+
+        return Duration.ofNanos((long) (duration.toNanos() / scalar));
```

Comment:
I feel similarly that floating point division is speculative. It's not clear to me that we are guaranteeing

```
long scalar = ...;
Duration byInt = divide(duration, scalar);
Duration byFloating = divide(duration, (double)scalar);
assertEquals(byInt, byFloating);
```
, even if we restrict scalar to values where the double value cleanly cast back into itself `scalar == (long)(double)scalar`.

Reply:
@cpwright and @rbasralian don't want to support this because it is not in the Java time lib.  Removed.